docs/sdks/uid2-sdk-ref-python.md

```diff
@@ -90,62 +97,101 @@ Available information returned through the SDK is outlined in the following tabl
    ```
+   <!-- Alternative to the above for EUID:
+   ```python
+   from uid2_client import EuidClientFactory
+   client = EuidClientFactory.create(base_url, auth_key, secret_key) 
+   ``` -->
 2. Refresh once at startup, and then periodically (recommended refresh interval is hourly):
 
    ```python
-   keys = client.refresh_keys()
+   client.refresh_keys()
    ```
 
 3. Senders: 
    1. Call the `encrypt` function. Then, if encryption succeeded, send the UID2 token to the receiver:
 
       ```python
-      from uid2_client import encrypt
-      from uid2_client.identity_scope import IdentityScope
-      
       try:
-         identity_scope = IdentityScope.UID2
-         encrypted_data = encrypt(raw_uid, identity_scope, keys)
-         #send encrypted_data to receiver
+         encrypted_data = client.encrypt(raw_uid)
+         # send encrypted_data to receiver
       except EncryptionError as err:
-        #check for failure reason
+        # check for failure reason
         print(err)
-      ``` 
-<!-- Alternative to the above for EUID:
-      from uid2_client import encrypt
-      from uid2_client.identity_scope import IdentityScope
-      
-        try:
-         identity_scope = IdentityScope.UID2  # or IdentityScope.EUID
-         encrypted_data = encrypt(raw_uid, identity_scope, keys)
-         #send encrypted_data to receiver
-      except EncryptionError as err:
-        #check for failure reason
-        print(err) -->
+      ```
 
 4. Receivers:
    1. Call the `decrypt` function. Then, if decryption succeeded, use the raw UID2:
     
       ```python
-      from uid2_client import decrypt
       try:
-        result = decrypt(ad_token, keys)
-        #use successfully decrypted result.uid2
+        result = client.decrypt(ad_token)
+        # use successfully decrypted result.uid2
       except EncryptionError as err:
-        #check for failure reason
+        # check for failure reason
         print(err)
       ```
 
+## Usage for Publishers
+
+1. Create an instance of Uid2PublisherClient
+
+   `client = Uid2PublisherClient(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY)`
+
+2. Call a function that takes the user's email address or phone number as input and generates a `TokenGenerateResponse` object. The following example uses an email address:
+
+   `token_generate_response = client.generate_token(TokenGenerateInput.from_email(emailAddress).do_not_generate_tokens_for_opted_out())`
+
+   >IMPORTANT: Be sure to call this function only when you have obtained legal basis to convert the user’s [directly identifying information (DII)](https://unifiedid.com/docs/ref-info/glossary-uid#gl-dii) to UID2 tokens for targeted advertising.
+
+   >`do_not_generate_tokens_for_opted_out()` applies `policy=1` in the [/token/generate](https://unifiedid.com/docs/endpoints/post-token-generate#token-generation-policy) call. Without this, `policy` is omitted to maintain backwards compatibility.
```

Comment:
>`do_not_generate_tokens_for_opted_out()` applies `policy=1` in the [POST /token/generate](../endpoints/post-token-generate#token-generation-policy) call. Without this, `policy` is omitted to maintain backwards compatibility.

Update to endpoint copy + link

Reply:
updated to relative url and changed link text to include POST

Reply:
TY!